app/src/main/java/com/keylesspalace/tusky/adapter/StatusDetailedViewHolder.java

```diff
@@ -152,38 +155,63 @@ protected void setupWithStatus(final StatusViewData.Concrete status, final Statu
+                    cardDescription.setVisibility(View.GONE);
+                } else {
+                    cardDescription.setVisibility(View.VISIBLE);
+                    if(TextUtils.isEmpty(card.getDescription())) {
+                        cardDescription.setText(card.getAuthorName());
+                    } else {
+                        cardDescription.setText(card.getDescription());
+                    }
+                }
 
                 cardUrl.setText(card.getUrl());
 
-                if (card.getWidth() > 0 && card.getHeight() > 0 && !TextUtils.isEmpty(card.getImage())) {
-                    cardImage.setVisibility(View.VISIBLE);
+                if (!TextUtils.isEmpty(card.getImage())) {
+
+                    RoundedCornersTransformation.CornerType cornertype;
 
                     if (card.getWidth() > card.getHeight()) {
                         cardView.setOrientation(LinearLayout.VERTICAL);
+
                         cardImage.getLayoutParams().height = cardImage.getContext().getResources()
                                 .getDimensionPixelSize(R.dimen.card_image_vertical_height);
                         cardImage.getLayoutParams().width = ViewGroup.LayoutParams.MATCH_PARENT;
                         cardInfo.getLayoutParams().height = ViewGroup.LayoutParams.MATCH_PARENT;
                         cardInfo.getLayoutParams().width = ViewGroup.LayoutParams.WRAP_CONTENT;
+                        cornertype = RoundedCornersTransformation.CornerType.TOP;
                     } else {
                         cardView.setOrientation(LinearLayout.HORIZONTAL);
+                        cardInfo.setMinimumHeight(cardImage.getContext().getResources()
+                                .getDimensionPixelSize(R.dimen.card_image_horizontal_width));
                         cardImage.getLayoutParams().height = ViewGroup.LayoutParams.MATCH_PARENT;
                         cardImage.getLayoutParams().width = cardImage.getContext().getResources()
                                 .getDimensionPixelSize(R.dimen.card_image_horizontal_width);
                         cardInfo.getLayoutParams().height = ViewGroup.LayoutParams.WRAP_CONTENT;
                         cardInfo.getLayoutParams().width = ViewGroup.LayoutParams.MATCH_PARENT;
+                        cornertype = RoundedCornersTransformation.CornerType.LEFT;
                     }
 
-                    cardView.setClipToOutline(true);
+                    int radius = cardImage.getContext().getResources()
+                            .getDimensionPixelSize(R.dimen.card_radius);
 
+                    cardView.setClipToOutline(true);
                     Glide.with(cardImage)
                             .load(card.getImage())
-                            .centerCrop()
+                            .transform(new CenterCrop(), new RoundedCornersTransformation(radius, 0, cornertype))
```

Comment:
I found no other way to get the rounded edge pixel-perfect than using this library unfortunately :(
The `setClipToOutline(true)` trick was off a pixel and looked weird.

Reply:
Ouch :( why it's so hard to just draw them correctly...